core/src/bundle/usBundleArchive_p.h

```diff
@@ -0,0 +1,171 @@
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+=============================================================================*/
+
+#ifndef USBUNDLEARCHIVE_P_H
+#define USBUNDLEARCHIVE_P_H
+
+#include "usChrono.h"
+
+#include <string>
+#include <memory>
+#include <vector>
+
+namespace us {
+
+class BundleResource;
+class BundleResourceContainer;
+struct BundleStorage;
+
+/**
+ * Class for managing bundle data.
+ */
+struct BundleArchive : std::enable_shared_from_this<BundleArchive>
+{
+
+  struct Data
+  {
+    long bundleId;
+    int64_t lastModified;
+    int32_t autostartSetting;
+  };
+
+  BundleArchive(const BundleArchive&) = delete;
+  BundleArchive& operator=(const BundleArchive&) = delete;
+
+  BundleArchive();
+
+  BundleArchive(
+      BundleStorage* storage,
+      std::unique_ptr<Data>&& data,
+      const std::shared_ptr<const BundleResourceContainer>& resourceContainer,
+      const std::string& resourcePrefix,
+      const std::string& location
+      );
+
+  /**
+   * Autostart setting stopped.
+   *
+   * @see BundleArchive#setAutostartSetting(String)
+   */
+  static const std::string AUTOSTART_SETTING_STOPPED; // = "stopped";
+
+  /**
+   * Autostart setting eager.
+   *
+   * @see BundleArchive#setAutostartSetting(String)
+   */
+  static const std::string AUTOSTART_SETTING_EAGER; // = "eager";
+
+  /**
+   * Autostart setting declared activation policy.
+   *
+   * @see BundleArchive#setAutostartSetting(String)
+   */
+  static const std::string AUTOSTART_SETTING_ACTIVATION_POLICY; // = "activation_policy";
+
+  bool IsValid() const;
+
+  /**
+   * Remove bundle archive from storage.
+   */
+  void Purge();
+
+  /**
+   * Get bundle identifier for this bundle archive.
+   *
+   * @return Bundle identifier.
+   */
+  long GetBundleId() const;
+
+  /**
+   * Get bundle location for this bundle archive.
+   *
+   * @return Bundle location.
+   */
+  std::string GetBundleLocation() const;
+
+  /**
+   * Get resource prefix in the shared resource container for this bundle archive.
+   *
+   * @return Resource prefix.
+   */
+  std::string GetResourcePrefix() const;
+
+  /**
+   * Get a BundleResource to named entry inside a bundle.
+   *
+   * @param path Entry to get reference to.
+   * @return BundleResource to entry.
+   */
+  BundleResource GetResource(const std::string& path) const;
+
+  /**
+   * Returns a list of all the paths to entries within the bundle matching
+   * the pattern.
+   *
+   * @param path
+   * @param filePattern
+   * @param recurse
+   * @return
+   */
+  std::vector<BundleResource> FindResources(const std::string& path, const std::string& filePattern,
+                                            bool recurse) const;
+
+  /**
+   * Get last modified timestamp.
+   */
+  Clock::time_point GetLastModified() const;
```

Comment:
Good catch. This is an omission. I will add a small test to check for correct timestamps. The last modified time is updated when bundles are installed, updated (not applicable yet) or uninstalled.